datajoint/s3.py

```diff
@@ -42,6 +42,12 @@ def get(self):
         self.object.download_fileobj(obj)
         return obj.getvalue()
 
-    def delete(self):
-        r = self.object.delete()
-        return r['ResponseMetadata']['HTTPStatusCode'] == 204
+
+def delete_all_except(except_list, bucket, aws_access_key_id, aws_secret_access_key, location, database, **_):
+    cred = (aws_access_key_id, aws_secret_access_key)
+    if cred not in sessions:
+        # cache sessions
+        session = boto3.Session(aws_access_key_id=aws_access_key_id,
+                                aws_secret_access_key=aws_secret_access_key)
+        sessions[cred] = session.resource('s3')
+    raise NotImplementedError
```

Comment:
via http://boto3.readthedocs.io/en/latest/guide/migrations3.html, raw boto delete example is:
```
# Boto 3
for key in bucket.objects.all():
    key.delete()
```
so basically this + checking against the file list should do it.
didn't fork/implement just yet since there is an open question about refactoring s3.py to zap the dj Bucket class & not 100% how this will map into this.